docfx_project/articles/bootstrapping-trust/part-02.a-primer-on-public-key-cryptography.md

```diff
@@ -0,0 +1,168 @@
+available public key to send messages only the private key holder can
+read. This property ensures that communication from the public key
+holder to the private key holder is being handled exclusively by the
+intended target, the private key holder.
+
+There is quite a bit of pressure to keep the private key extremely safe.
+Whoever holds the private key, has a guaranteed identity that is tied to
+and verifiable by the public key. It is verifiable because if one can
+use the public key to encrypt data, only the private key holder can
+decrypt it. This fact means that data can be encrypted and sent that
+coordinates on an additional secret. Since only the private key holder
+can decrypt the data to see this second level secret, future
+communication can use the new secret to encrypted and verify traffic in
+both directions. This additional exchange is roughly how part of the TLS
+negotiation works for HTTPs. TLS, and by proxy HTTPS, use other
+technologies and strategies to be an incredible security proposition.
+
+#### Private Key Encryption
+
+For private key encryption, the same principles as with public key
+encryption with the roles reversed. The private key encrypts data only
+the public key can decrypt. On the surface, this seems absurd. When the
+server encrypts data with its private key, the public key can decrypt
+it. The public key is not protected and expected to be widely available.
+It seems as if private key encryption is nearly useless as everyone can
+read it!
+
+Except it isn't. Private key encryption verifies the identity of the
+private key holder. The public key cannot interact with anyone else.
+Additionally, this property allows us to generate encrypted data that
+could only have come from the private key holder. If that data happens
+to be small and describe another document, we call that a "digital
+signature" or "signature" for short.
+
+### Digital Signatures
+
+Digital signatures are similar to handwritten ones used to sign legal
+documents and checkbooks, but with a significant advantage. They
+validate a document has not been altered since it was signed. With
+today's computer's graphical abilities, the nefarious can forge images
+and handwritten signatures. That puts handwritten signatures at a
+significant disadvantage. So how does this work?
+
+The data that will be signed can be anything. What it represents is not
+important. It can be text, JSON, an image, a PDF, or anything at all!
+That data processed by a one-way hashing algorithm, such as SHA-256.
+This process is idempotent, meaning running it repeatedly on the same
+data results and results in the same output. The output of this process
+is a hash, a string of characters that uniquely identifies the input
+data. With sufficiently large input data, the hash is much shorter than
+the input data as the hash size is usually fixed length.
+
+For example, here is the Ziti logo:
+
+![Image of a simple mesh](./images/ziti.png)
```

Comment:
image is missing. did you mean ziti-system.png?

Reply:
actually this needs a moment. The leading sentence says: "For example, here is the Ziti logo" but the image hyperlink uses the text of "Image of a simple mesh" which seems like ziti-system.png is the right one.

Reply:
There was an image missing: added and fixed text.